core/src/main/scala/cats/syntax/transLift.scala

```diff
@@ -6,5 +6,24 @@ trait TransLiftSyntax {
 }
 
 final class TransLiftOps[M[_], A](val ma: M[A]) extends AnyVal {
-  def liftT[MT[_[_],_]](implicit TL: TransLift[MT, M]): MT[M,A] = TL.liftT(ma)
+  def liftT[MT[_[_],_]](implicit extract: TLExtract[MT, M]): MT[M, A] = extract.TL.liftT(ma)(extract.TC)
 }
+
+trait TLExtract[MT[_[_], _], M[_]] {
+  val TL: TransLift[MT]
+  val TC: TL.TC[M]
```

Comment:
I get uneasy too.  :-)  In this case, it's perfectly safe since we're using the trait as a pure interface (no definitions) and we aren't mixing it into anything, so there are no eval orders to get screwed up.